src/viam/app/app_client.py

```diff
@@ -324,6 +347,27 @@ def proto(self) -> RobotPartHistoryEntryPB:
         )
 
 
+class APIKeyAuthorization:
+    """A class with the necessary authorization data for creating an API key.
+
+    Use this class when constructing API key authorizations to minimize the risk of malformed or missing data.
+    """
+
+    def __init__(self, role: str, resource_type: str, resource_id: str):
```

Comment:
You might want to make these type hints `Union[Literal["owner"], Literal["operator"]]` and ``Union[Literal["organization"], Literal["location"], Literal["robot"]]`